src/libs/ontouml2db/factory/Factory.ts

```diff
@@ -0,0 +1,316 @@
+        relation.id,
+        sourceNode,
+        this.getCardinality(sourceCardinality),
+        targetNode,
+        this.getCardinality(targetCardinality),
+      );
+
+      sourceNode.addRelation(newRelation);
+      targetNode.addRelation(newRelation);
+
+      this.graph.addRelation(newRelation);
+    });
+  }
+
+  getCardinality(cardinality: string): Cardinality {
+    let lowerCardinality = this.getLowerboundCardinality(cardinality);
+    let uppetCardinality = this.getUpperboundCardinality(cardinality);
+
+    if (lowerCardinality == 0) {
+      if (uppetCardinality == 1) return Cardinality.C0_1;
+      else return Cardinality.C0_N; // 0..2, 0..3, ..., 0..*
+    } else {
+      if (lowerCardinality == 1) {
+        if (uppetCardinality == 1) return Cardinality.C1;
+        else return Cardinality.C1_N; // 1..2, 1..3, ..., 1..*
+      } else {
+        return Cardinality.C1_N;
+      }
+    }
+  }
+
+  getLowerboundCardinality(cardinality: string): number {
+    const cardinalities = cardinality.split('..');
+    const lowerbound = cardinalities[0];
+
+    return lowerbound === '*' ? 99999 : Number(lowerbound);
+  }
+
+  getUpperboundCardinality(cardinality: string): number {
+    const cardinalities = cardinality.split('..');
+    const upperbound = cardinalities[1] || cardinalities[0];
+
+    return upperbound === '*' ? 99999 : Number(upperbound);
+  }
+
+  /********************************************************************
+   ** puts the generalizations
+   *********************************************************************/
+  putGeneralizaitons(): void {
+    let newGeneralization: IGraphGeneralization;
+    let generalizationNode: INode;
+    let specializationNode: INode;
+    let generalizations = this.modelManager.rootPackage.getAllContentsByType([
+      OntoUMLType.GENERALIZATION_TYPE,
+    ]) as IGeneralization[];
+
+    generalizations.forEach((gen: IGeneralization) => {
+      generalizationNode = this.graph.getNodeById(gen.general.id);
+      specializationNode = this.graph.getNodeById(gen.specific.id);
+
+      newGeneralization = new GraphGeneralization(
+        gen.id,
+        generalizationNode,
+        specializationNode,
+      );
+
+      generalizationNode.addGeneralization(newGeneralization);
+      specializationNode.addGeneralization(newGeneralization);
+
+      this.graph.addGeneralization(newGeneralization);
+    });
+  }
+
+  /********************************************************************
+   ** puts the generalization sets
+   *********************************************************************/
+  putGeneralizationSets(): void {
+    let newGeneralizationSet: IGraphGeneralizationSet;
+    let graphGeneralization: IGraphGeneralization;
+    let generalizationSets = this.modelManager.rootPackage.getAllContentsByType(
+      [OntoUMLType.GENERALIZATION_SET_TYPE],
+    ) as IGeneralizationSet[];
+
+    //informar para cada generalização que ela pertence a um conjunto de generalização
```

Comment:
Please make sure all comments are in English.

Reply:
Done.